File: backend/src/main/java/com/grad/social/model/chat/response/ChatResponse.java
package com.grad.social.model.chat.response;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
import lombok.Data;

import java.time.Instant;

@Data
public class ChatResponse {
    @JsonSerialize(using = ToStringSerializer.class)
    private Long chatId;
    private String name;
    private byte[] chatPicture;
    private String lastMessage;
    private Instant lastMessageTime;
    private String messageType;
    private Long unreadCount;
    private int chatMembersNumber;
    private int onlineRecipientsNumber;
    private boolean isPinned;
    private boolean isMuted;
    private boolean isGroup;
}

File: frontend/app/src/app/components/chatting/models/chat-response.ts

export interface ChatResponse {
  chatId: string;
  name?: string;
  chatPicture: string; // Base64 encoded
  lastMessage?: string;
  lastMessageTime: string | number | Date;
  onlineRecipientsNumber: number; // fix here!
  unreadCount: number;
  pinned: boolean;
  muted: boolean;
  messageType: string;
  chatMembersNumber: number;
  group
  : boolean;
}





